test/views/declaration/ConsigneeDetailsViewSpec.scala

```diff
@@ -32,18 +32,58 @@ import views.html.declaration.consignee_details
 import views.tags.ViewTest
 
 @ViewTest
-class ConsigneeDetailsViewSpec extends UnitViewSpec with CommonMessages with Stubs {
+class ConsigneeDetailsViewSpec extends UnitViewSpec with CommonMessages with Stubs with Injector {
 
   val form: Form[ConsigneeDetails] = ConsigneeDetails.form()
-  val consigneeDetailsPage = new consignee_details(mainTemplate)
-  private def createView(form: Form[ConsigneeDetails] = form)(implicit request: JourneyRequest[_]): Document = consigneeDetailsPage(Mode.Normal, form)
+  val consigneeDetailsPage = instanceOf[consignee_details]
+  private def createView(form: Form[ConsigneeDetails] = form)(implicit request: JourneyRequest[_]): Document =
+    consigneeDetailsPage(Mode.Normal, form)(request, messages)
+
+  val allFields = Seq("fullName", "addressLine", "townOrCity", "addressLine", "postCode", "country")
+  val validAddress = Address("Marco Polo", "Test Street", "Leeds", "LS18BN", "England")
+  val emptyAddress = Address("", "", "", "", "")
+  val randomAddress = Address(
+    TestHelper.createRandomAlphanumericString(71),
+    TestHelper.createRandomAlphanumericString(71),
+    TestHelper.createRandomAlphanumericString(71),
+    TestHelper.createRandomAlphanumericString(71),
+    TestHelper.createRandomAlphanumericString(71)
+  )
+
+  def createViewWithAddressError(address: Address)(implicit request: JourneyRequest[_]): Document = {
+    val view = createView(
+      ConsigneeDetails
+        .form()
+        .fillAndValidate(ConsigneeDetails(EntityDetails(None, Some(address))))
+    )
+    view must haveGovukGlobalErrorSummary
+    view
+  }
+
+  def assertAllAddressValidation(errorKey: String, fieldName: String, address: Address, fields: Seq[String] = allFields)(
+    implicit request: JourneyRequest[_]
+  ): Unit = {
+
+    val view = createViewWithAddressError(address)
+    fields.filterNot(_ == fieldName).foreach { key =>
+      view must containErrorElementWithTagAndHref("a", s"#details_address_$key")
+      view must containErrorElementWithMessage(s"supplementary.address.$key.$errorKey")
+    }
+  }
```

Comment:
Is it worth to create this method?

This part: `fields.filterNot(_ == fieldName)` I don't understand why we need all fields except the one that is passed to this.

I think this assertion can be made directly inside the code using the second method, where we can iterate through the sequence and invoke `assertAddressValidation`.

If you want to stay with this method it can be like `fields.filterNot(_ == fieldName).foreach(assertAddressValidation(_))`

There is the same logic in both methods, but one of them iterate through the sequence

Reply:
In general you can add view as an argument. In that case you should be able to stay with one method and iterate through the sequence in the specific test

Reply:
Regarding this fragment `This part: fields.filterNot(_ == fieldName) I don't understand why we need all fields except the one that is passed to this.`,  I used it to replace a repetitive code that was present in the previous version. I preserved the intent of the previous test to verify if multiple errors are shown correctly.